runtime/src/snapshot_utils.rs

```diff
@@ -996,6 +1540,75 @@ pub fn bank_to_snapshot_archive<P: AsRef<Path>, Q: AsRef<Path>>(
     Ok(package.tar_output_file)
 }
 
+/// Convenience function to create an incremental snapshot archive out of any Bank, regardless of
+/// state.  The Bank will be frozen during the process.
+///
+/// Requires:
+///     - `bank` is complete
+///     - `bank`'s slot is greater than `full_snapshot_slot`
+pub fn bank_to_incremental_snapshot_archive<P: AsRef<Path>, Q: AsRef<Path>>(
+    snapshot_path: P,
+    bank: &Bank,
+    full_snapshot_slot: Slot,
+    snapshot_version: Option<SnapshotVersion>,
+    snapshot_package_output_path: Q,
+    archive_format: ArchiveFormat,
+    thread_pool: Option<&ThreadPool>,
+    maximum_snapshots_to_retain: usize,
+) -> Result<PathBuf>
+where
+    P: fmt::Debug,
+{
+    assert!(bank.is_complete());
+    assert!(bank.slot() > full_snapshot_slot);
+
+    bank.squash(); // Bank may not be a root
+    bank.force_flush_accounts_cache();
+
+    // bprumo TODO: Cleaning still needs to be figured out.  My understanding is that we cannot
+    // clean past the last full snapshot slot here, but more testing needs to be done.  For now,
+    // I'm not cleaning at all when creating an incremental snapshot, since a full snapshot must
+    // have been created first, which would have performed a clean then.  Below are the different
+    // `clean_accounts()` options:
+    //      bank.clean_accounts(true, false);
+    //      bank.clean_accounts_up_to_slot(Some(full_snapshot_slot), false);
+    //      bank.clean_accounts_up_to_slot(Some(full_snapshot_slot.saturating_sub(1)), false);
```

Comment:
Note that these three are all effectively the same, if `AccountsDb::last_full_snapshot_slot` is correct.